dev/bots/test.dart

```diff
@@ -301,6 +301,20 @@ Future<void> _runIntegrationToolTests() async {
       .map<String>((FileSystemEntity entry) => path.relative(entry.path, from: toolsPath))
       .where((String testPath) => path.basename(testPath).endsWith('_test.dart')).toList();
 
+  // Make sure devtools is ready first, because that might take a while and we don't
+  // want any of the tests to time out while they themselves try to activate devtools.
+  final Map<String, String> pubEnvironment = <String, String>{
+    'FLUTTER_ROOT': flutterRoot,
+  };
+  if (Directory(pubCache).existsSync()) {
+    pubEnvironment['PUB_CACHE'] = pubCache;
+  }
+  await runCommand(
+    pub,
+    <String>['global', 'activate', 'devtools'],
```

Comment:
@jonahwilliams mentioned today that there can be a risk of inconsistent state if two instances of `pub` are running at the same time in the same pub cache. @jonahwilliams could this invocation race with the invocation in the flutter_tool? Also, due to a recent break, we'd like to pin devtools: https://github.com/flutter/flutter/pull/86124

```suggestion
    <String>['global', 'activate', 'devtools', '2.4.0'],
```

Reply:
As long as this command is run before we start any tools or other tests it should be OK. In the previous case pub global activate was running close to the same time as the flutter tool was started along side the host side driver test running update

Reply:
I'll make this pin the pubtools version (but with a constant).

As Jonah says, this should be pretty safe. Nothing else should be running on the VM at this point.

Reply:
I've got a PR going with a constant here https://github.com/flutter/flutter/pull/86130

Reply:
Added a commit that pins with a `.version` file like other pins.